src/Controller/IndexController.php

```diff
@@ -416,13 +422,11 @@ protected function getTempPath($tempDir = null)
             return $this->tempPath;
         }
         if (!isset($tempDir)) {
-            $config = $this->config;
-            if (!isset($config['temp_dir'])) {
+            if (!isset($this->tempDir)) {
                 throw new ConfigException('Missing temporary directory configuration');
             }
-            $tempDir = $config['temp_dir'];
         }
-        $this->tempPath = tempnam($tempDir, 'omeka');
+        $this->tempPath = tempnam($this->tempDir, 'omeka');
```

Comment:
This would seem to break a passed in $tempDir by never using it. I don't know if we actually use that capability in here but, it's a change in behavior so probably best avoided.